podvm/README.md

```diff
@@ -0,0 +1,134 @@
+
+The next sections describe that process in details. Note that although the following examples use docker, it can be carried out with podman too.
+
+## Building a builder image
+
+The builder image packages the cloud-api-adaptor and Kata Containers sources as well as the softwares to build
+the binaries (e.g. *kata-agent* and *agent-protocol-forwarder*) that should be installed in the podvm image.
+
+The builder image is agnostic to cloud providers in the sense that one can be used to build for multiple providers, however it is
+dependent of the Linux distribution the image is built for. Therefore, in this directory you will find dockerfiles to each supported distributions, which are currently Ubuntu 20.04 ([Dockerfile.podvm_builder](./Dockerfile.podvm_builder)), CentOS Stream 8 ([Dockerfile.podvm_builder.centos](./Dockerfile.podvm_builder.centos)), and RHEL 8.7 ([Dockerfile.podvm_builder.rhel](./Dockerfile.podvm_builder.rhel)).
+
+As an example, to build the builder image for Ubuntu, run:
+
+```
+$ docker build -t podvm_builder -f Dockerfile.podvm_builder .
+```
+
+Use `--build-arg` to pass build arguments to docker to overwrite default values if needed. Following are the arguments
+currently accepted:
+
+|Argument|Default value|Description|
+|--------|-------------|-----------|
+|CAA\_SRC |https://github.com/confidential-containers/cloud-api-adaptor | The cloud-api-adaptor source repository |
+|CAA\_SRC\_BRANCH|staging| cloud-api-adaptor repository branch |
+|KATA\_SRC | https://github.com/kata-containers/kata-containers | The Kata Containers source repository |
+|KATA\_SRC\_BRANCH | CCv0 | The Kata Containers repository branch |
+|GO\_VERSION | 1.18.7 | Go version |
+|PROTOC\_VERSION | 3.11.4 | [Protobuf](https://github.com/protocolbuffers/protobuf) version |
+|RUST\_VERSION | 1.66.0 | Rust version |
+
+As it can be noted on the table above the cloud-api-adaptor repository is cloned within the builder image, so rather than
+copying the local source tree, it will be used the upstream source. But if you want to test local changes then you should:
+
+* Push the changes to elsewhere (e.g. https://github.com/$USER/cloud-api-adaptor/tree/my-changes-in-a-branch).
+* Overwrite the *CAA_SRC* and *CAA_SRC_BRANCH* arguments as shown below:
+
+```
+$ docker build -t podvm_builder \
+	--build-arg CAA_SRC=https://github.com/$USER/cloud-api-adaptor \
+	--build-arg CAA_SRC_BRANCH=my-changes-in-a-branch \
+	-f Dockerfile.podvm_builder .
+```
+
+## Building the podvm qcow2 image
+
+The process to build the podvm container image will effectively compile binaries and create the qcow2 file.
+
+In order to build the podvm image there should be used the corresponding dockerfile of the Linux distro which the builder image (see the [previous step](#building-a-builder-image)) was built. For example, if the builder image was built with *Dockerfile.podvm_builder.DISTRO* then you should use the *Dockerfile.podvm.DISTRO* to build the podvm image.
```

Comment:
"In order to build the podvm image there should be used the corresponding dockerfile of the Linux distro which the builder image was built" => "In order to build the podvm image you should be using the corresponding dockerfile of the Linux distro for which the builder image was built"